store/querier.go

```diff
@@ -52,43 +61,74 @@ func NewUtxoQuerier(utxoStore UTXOStore) sdk.Querier {
 				return nil, sdk.ErrInternal("serialization error")
 			}
 			return data, nil
-
+		case QueryTransactionOutput:
+			if len(path) != 2 {
+				return nil, sdk.ErrUnknownRequest("expected txo/<position>")
+			}
+			pos, e := plasma.FromPositionString(path[1])
+			if e != nil {
+				return nil, sdk.ErrInternal("position decoding error")
+			}
+			output, err := queryTransactionOutput(ctx, outputStore, pos)
+			if err != nil {
+				return nil, err
+			}
+			txo := TransactionOutput{}
+			data, e := json.Marshal(txo)
+			if e != nil {
+				return nil, sdk.ErrInternal("serialization error")
+			}
+			return data, nil
+		case QueryTx:
+			if len(path) != 2 {
+				return nil, sdk.ErrUnknownRequest("expected tx/<hash>")
+			}
+			tx, ok := outputStore.GetTx(ctx, []byte(path[1]))
+			if !ok {
+				return nil, ErrTxDNE(fmt.Sprintf("no transaction exists for the hash provided: %x", []byte(path[1])))
+			}
+			data, e := json.Marshal(tx)
+			if e != nil {
+				return nil, sdk.ErrInternal("serialization error")
+			}
+			return data, nil
 		default:
 			return nil, sdk.ErrUnknownRequest("unregistered endpoint")
 		}
 	}
 }
 
-func queryBalance(ctx sdk.Context, utxoStore UTXOStore, addr common.Address) (*big.Int, sdk.Error) {
-	iter := sdk.KVStorePrefixIterator(utxoStore.KVStore(ctx), addr.Bytes())
-	total := big.NewInt(0)
-	for ; iter.Valid(); iter.Next() {
-		utxo, ok := utxoStore.GetUTXOWithKey(ctx, iter.Key())
-		if !ok {
-			return nil, sdk.ErrInternal("failed utxo retrieval")
-		}
-
-		if !utxo.Spent {
-			total = total.Add(total, utxo.Output.Amount)
-		}
+func queryBalance(ctx sdk.Context, outputStore OutputStore, addr common.Address) (*big.Int, sdk.Error) {
+	acc, ok := outputStore.GetWallet(ctx, addr)
+	if !ok {
+		return nil, ErrWalletDNE(fmt.Sprintf("no wallet exists for the address provided: 0x%x", addr))
 	}
 
-	return total, nil
+	return acc.Balance, nil
 }
 
-func queryInfo(ctx sdk.Context, utxoStore UTXOStore, addr common.Address) ([]UTXO, sdk.Error) {
-	var utxos []UTXO
-	iter := sdk.KVStorePrefixIterator(utxoStore.KVStore(ctx), addr.Bytes())
-	for ; iter.Valid(); iter.Next() {
-		utxo, ok := utxoStore.GetUTXOWithKey(ctx, iter.Key())
-		if !ok {
-			return nil, sdk.ErrInternal("failed utxo retrieval")
-		}
+func queryInfo(ctx sdk.Context, outputStore OutputStore, addr common.Address) ([]OutputInfo, sdk.Error) {
+	acc, ok := outputStore.GetWallet(ctx, addr)
+	if !ok {
+		return nil, ErrWalletDNE(fmt.Sprintf("no wallet exists for the address provided: 0x%x", addr))
+	}
+	return outputStore.GetUnspentForWallet(ctx, acc), nil
+}
+
+func queryTransactionOutput(ctx sdk.Context, outputStore OutputStore, pos plasma.Position) (TransactionOutput, sdk.Error) {
```

Comment:
this needs to be finished, tx.InputAddress() isn't a func in Transaction nor do I think it can be. I think we need to add InputTx to the wrapped Transaction struct in store/ then query for that here and then set input addresses in TransactionOutput